sympy/solvers/solveset.py

```diff
@@ -1142,13 +1142,135 @@ def _is_exponential(f, symbol):
+    to a single instance of `log`.
+
+    Parameters
+    ==========
+
+    f : Expr
+        The logarithmic equation to be solved as an instance of `Eq`
+
+    symbol : Symbol
+        The variable in which the equation is solved
+
+    Returns
+    =======
+
+    An improved equation containing a single instance of log.
+
+    Examples
+    ========
+
+    >>> from sympy import symbols, S, log, Eq
+    >>> from sympy.solvers.solveset import \
+    ... _solve_logarithm_reducable_to_single_instance as solve_log
+    >>> x = symbols('x')
+    >>> f = Eq(log(x - 3) + log(x + 3), 0)
+    >>> solve_log(f, x)
+    log((x - 3)*(x + 3))
+
+    * Proof of correctness
+
+    A logarithm is another way to write exponent and is defined by
+
+    .. math:: {log_b x} = y \enspace if \enspace b^y = x
+
+    When one side of the equation contains a single logarithm, the
+    equation can be solved by rewriting the equation as an equivalent
+    exponential equation as defined above. But if one side contains
+    more than one logarithm, we need to use the properties of logarithm
+    to condense it into a single logarithm.
+
+    Take for example
+
+    .. math:: log(2*x) - 15 = 0
+
+    contains single logarithm, therefore we can directly rewrite it to
+    exponential form as
+
+    .. math:: x = exp(15)/2
+
+    But if the equation has more than one logarithm as
+
+    .. math:: log(x - 3) + log(x + 3) = 0
+
+    we use log identities to convert into a reduced form
+
+    Using,
+
+    .. math:: log(a) + log(b) = log(a*b)
+
+    therefore the equation becomes,
+
+    .. math:: log((x - 3)*(x  + 3))
+
+    This equation contains one logarithm and can be solved by rewriting
+    to exponents.
+    """
+
+    lhs, rhs = f.lhs, f.rhs
+    new_lhs = logcombine(lhs, force=True)
+    new_f = new_lhs - rhs
+
+    return new_f
+
+
+def _is_logarithm_reducable_to_single_instance(f, symbol):
+    r"""
+    Helper to check if the given equation can be reduced to a
+    single instance of log containing the variable to be solved.
+
+
+    Parameters
+    ==========
+
+    f : Expr
+        The equation to be checked as an instance of `Eq`
+
+    symbol : Symbol
+        The variable in which the equation is checked
+
+    Returns
+    =======
+
+    `True` if the equation is reducable otherwise `False`.
+
+    Examples
+    ========
+
+    >>> from sympy import symbols, tan, log, Eq
+    >>> from sympy.solvers.solveset import \
+    ... _is_logarithm_reducable_to_single_instance as check
+    >>> x = symbols('x')
+    >>> check(Eq(log(x + 2) - log(x + 3), 0), x)
+    True
+    >>> check(Eq(tan(log(2*x)), 0), x)
+    False
+
+    * Philosophy behind the helper
+
+    The function uses `logcombine` to see whether the equation
+    gets reduced to a single instance of `log`.
+    """
+    lhs, rhs = f.lhs, f.rhs
+    new_lhs = logcombine(lhs, force=True)
+    if new_lhs is not lhs:
+        if isinstance(new_lhs, log) and symbol in new_lhs.free_symbols:
```

Comment:
why do you require that the new lhs contains symbol? What happens if you pass the constant back as the new lhs?

Reply:
It won't be a problem. I just added checking `symbol` considering your comment [here](https://github.com/sympy/sympy/pull/14736#discussion_r198129763).

Reply:
I suspect you should ignore the fact that it may no longer have the symbol. The equation came to you because it was transcendental; this function will tell transolve that it is can be solved and the log solver will send it back to `_solveset` in its new form -- nontranscendental -- and sovleset figures out what to do with it.

Reply:
okay, I will remove it.